[New Entry] added deque file for csharp (#5195)
Codecademy/docs: content/c-sharp/concepts/deque/deque.md

---
Title: 'Deque'

> Description: 'A deque is a type of data structure that allows insert and delete elements at both ends.'

Reviewer: ```suggestion
Description: 'A deque is a type of data structure that allows insertion and deletion of elements at both ends.'
```
Improved readability.

Subjects:
  - 'Computer Science'
  - 'Code Foundations'
Tags:

>   - 'Deques'
>   - 'Data Structures'
>   - 'Data Types'

Reviewer: ```suggestion
  - 'Deques'
  - 'Data Structures'
  - 'Data Types'
```
Include only those tags that are already present in the[ tags.md list](documentation/tags.md), and if you want the relevant tag then you can add one. Here `Deque` is not already available in the tags.md list, you can add it.

> Tags:
>   - 'Deques'
>   - 'Data Structures'
>   - 'Data Types'

Reviewer: ```suggestion
Tags:
  - 'Deques'
  - 'Data Structures'
  - 'Data Types'
```
Add `Deques` to the tags list.

CatalogContent:

>   - 'learn-c-sharp'
>   - 'paths/computer-science'

Reviewer: ```suggestion
  - 'learn-c-sharp'
  - 'paths/computer-science'
```
Include only those paths that are already present in the [catalog-content.md](documentation/catalog-content.md) and if you want the relevant course/paths to be included then you can add one. Here `learn-c-sharp` is not already available in the catalog-content.md file you can add it.

---

> A **deque** is a data structure that allows elements to be added or removed from both ends, making it more versatile than a traditional queue or stack. In `C#` there is no build-in `deque` but it can be impleted using `LinkedList` and `List` classes `C#`.

Reviewer: ```suggestion
A **deque** is a data structure that allows elements to be added or removed from both ends, making it more versatile than a traditional queue or stack. In `C#`, there is no built-in `deque` but it can be impleted using `LinkedList` and `List` classes.
```
Corrected some punctuation marks.
